code/src/components/TaskList.js

```diff
@@ -0,0 +1,252 @@
+
+    @media screen and (min-width: 1024px) {
+        font-size: 1rem;
+    }
+`
+
+const DeleteButton = styled.button`
+    position: absolute;
+    top: 1.5rem;
+    right: 1.5rem;
+    appearance: none;
+    border: none;
+    cursor: pointer;
+    background: linear-gradient(220deg, #c56bf9 0%, #e3c5f4 80%);
+    font-size: 0.8rem;
+    color: #000000ec;
+    border-radius: 50%;
+    width: 25px;
+    height: 25px;
+    box-shadow: 0px 3px 5px 0px #c56bf9;
+
+    &:hover {
+        transform: translateY(-5px);
+        box-shadow: 0px 5px 8px 2px #c56bf9d0;
+        transition: 0.4s ease-in-out;
+    }
+
+    @media screen and (min-width: 1024px) {
+        top: 2rem;
+        right: 2rem;
+        width: 35px;
+        height: 35px;
+        font-size: 1rem;
+    }
+`
+
+const TaskList = ({ allTasks, uncompletedTasks, completedTasks }) => {
+    const taskList = useSelector((state) => state.taskitems.taskitems)
+    const uncompletedTaskList = taskList.filter(task => task.complete === false)
+    const completedTaskList = taskList.filter(task => task.complete === true)
+    const dispatch = useDispatch()
+
+    const onTaskToggle = (taskId) => {
+        dispatch(taskitems.actions.toggleTaskItem(taskId))
+    } 
+
+    const onTaskDelete = (id) => {
+        dispatch(taskitems.actions.deleteTaskItem(id))
+    }
+
+    return (
+        <>
+        {taskList.length > 0 && <TaskFilter />}
+        {allTasks && 
+            <StyledTaskList>
+                {taskList.map((task) => (
+                    <TaskItem key={task.id}>
+                        <TaskTitle>{task.name}</TaskTitle>
+
+                        <TaskDetails>
+                            <label>
+                                <input
+                                type="checkbox"
+                                checked={task.complete}
+                                onChange={() => onTaskToggle(task.id)}
+                                />
+                                {task.complete && <CheckSpan>Done!</CheckSpan>}
+                            </label>
+
+                            <TaskTag
+                            className={task.priority === "Urgent" ? "urgent-tag"
+                            : task.priority === "Normal" ? "normal-tag"
+                            : task.priority === "Low"
+                            ? "low-tag" : ""}>
+                                {task.priority}
+                            </TaskTag>
+                        </TaskDetails>
+                        
+                        <TaskDate>Created <Moment fromNow>{task.date}</Moment></TaskDate>
+
+                        <DeleteButton onClick={() => onTaskDelete(task.id)}>
+                                <span role="img" aria-label="emoji">
+                                    ✘
+                                </span>
+                        </DeleteButton>
+                    </TaskItem>
+                ))}
+            </StyledTaskList> 
+        }
+
```

Comment:
Below you are repeating the same code depending how you want to filter them (all/completed/uncompleted). Instead of this you could set outside of the return with if / else statements that what tasklist will you map over (depending on what filter button you clicked).

Reply:
True, a lot of repetition here, thank you for the advice, will try it out! 🙏😊